### PR TITLE
add missing guards

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by SimonCropp** [comment]
@danielmarbach @andreasohlund there is one broken test in this PR that i dont know what to do about 
https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core.Tests/Handlers/MessageHandlerRegistryTests.cs#L25

it seems to be a cross between an integration test and a unit test. the guard is correct and we should not accept null on that API. so what do we do with the test?

**Comment by andreasohlund** [comment]
Just pass some fake values to make it pass?

**Comment by SimonCropp** [comment]
@andreasohlund i tried that. it then failed on the asserts. do u want o have a try?

**Comment by danielmarbach** [comment]
@SimonCropp @andreasohlund I'll handle this

File: src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs [hunk 1]
```diff
@@ -136,11 +136,12 @@ bool MethodCallSelf(MethodDefinition method)
 
         static void WriteMethod(MethodDefinition method, TextWriter writer)
         {
-            writer.WriteLine("\r\n" + method.DeclaringType.Name + "." + method.Name);
-            var instruction = method.Body.Instructions.FirstOrDefault(x => x.SequencePoint != null);
+            writer.WriteLine($"\r\n{method.DeclaringType.Name}.{method.Name}");
+            var instruction = method.Body.Instructions.FirstOrDefault(x => x.SequencePoint != null && x.SequencePoint.StartLine != 16707566);
```
Comment (danielmarbach) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: Could you document this magic constant?

**Comment by danielmarbach** [comment]
@SimonCropp I think the test is now fixed

File: src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs (hunk 1 above)
Comment (andreasohlund) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: Is it this http://stackoverflow.com/questions/21839237/how-to-limit-0xfeefee-line-numbers-in-pdb-files ?

Comment (SimonCropp) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: yep that is it. i will add a comment

Comment (SimonCropp) [inline src/NServiceBus.Core.Tests/ArgumentExceptionTests.cs]: added comment

**Pull request merged**